src/io/io.ts

```diff
@@ -89,27 +90,35 @@ export async function retypeFile(file: File): Promise<File> {
 
 export async function extractArchivesRecursively(
   files: File[]
-): Promise<File[]> {
+): Promise<FileEntry[]> {
   const results = await Promise.all(
     files.map(async (file) => {
       if (file.type === 'zip') {
```

Comment:
I think this also needs to be `ARCHIVE_FILE_TYPES.has(file.type)`, just to be sure.

Reply:
Done